turbopack/crates/turbopack-core/src/issue/mod.rs

```diff
@@ -512,6 +542,76 @@ impl IssueSource {
     }
 }
 
+async fn source_pos(
+    source: Vc<Box<dyn Source>>,
+    origin: Vc<FileSystemPath>,
+    start: SourcePos,
+    end: SourcePos,
+) -> Result<Option<(Vc<Box<dyn Source>>, SourcePos, SourcePos)>> {
+    let Some(generator) = Vc::try_resolve_sidecast::<Box<dyn GenerateSourceMap>>(source).await?
+    else {
+        return Ok(None);
+    };
+
+    let srcmap = generator.generate_source_map();
+    let Some(srcmap) = *srcmap.await? else {
+        return Ok(None);
+    };
+
+    // let Some(srcmap) = *convert_to_turbopack_source_map(srcmap, origin).await? else {
+    //     return Ok(None);
+    // };
+
+    let find = |line: usize, col: usize| async move {
+        let token = srcmap.lookup_token(line, col).await?;
+
+        match &*token {
+            crate::source_map::Token::Synthetic(t) => Ok::<_, anyhow::Error>((
+                t.guessed_original_file.clone(),
+                SourcePos {
+                    line: t.generated_line as _,
+                    column: t.generated_column as _,
+                },
+            )),
+            crate::source_map::Token::Original(t) => Ok((
+                Some(t.original_file.clone()),
+                SourcePos {
+                    line: t.original_line as _,
+                    column: t.original_column as _,
+                },
```

Comment:
Also lookup sourcesContent from source map here.

SourceMap need to be the result of convert_to_turbopack_source_map so sourceContent contains something.

Create a VirtualSource to expose that source code